cermine-impl/src/main/java/pl/edu/icm/cermine/ContentExtractor.java

```diff
@@ -847,20 +844,122 @@ public static void main(String[] args) throws ParseException, AnalysisException,
                 printException(ex);
             } catch (TimeoutException ex) {
                 printException(ex);
+            }catch (IOException ex) {
+                printException(ex);
+            }catch (Exception ex) {
+                printException(ex);
             } finally {
                 if (extractor != null) {
+                    extractor.printProgress(total.intValue());
```

Comment:
I think the progress shouldn't be printed by the task itself, but by the code executing the tasks in `main()`. If you used `CompletionService` instead of `ExecutorService`, you could easily monitor the progress by calling `take()` on the service.